internal/k8s/controllers/http_route_controller.go

```diff
@@ -53,5 +60,126 @@ func (r *HTTPRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
+	// TODO: Is there a safer way I could typecheck this with
+	// object.GetObjectKind() or something before casting?
+	refPolicy := *object.(*gateway.ReferencePolicy)
+	r.Log.Info("event for ReferencePolicy", "object", refPolicy)
+
+	routes := r.getRoutesAffectedByReferencePolicy(refPolicy)
+	requests := []reconcile.Request{}
+
+	for _, route := range routes {
+		requests = append(requests, reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      route.Name,
+				Namespace: route.Namespace,
+			},
+		})
+	}
+
+	return requests
+}
+
+func (r *HTTPRouteReconciler) getRoutesAffectedByReferencePolicy(refPolicy gateway.ReferencePolicy) []gateway.HTTPRoute {
+	matches := []gateway.HTTPRoute{}
+
+	// toSelectors := []fields.Selector{}
+	// for _, to := range refPolicy.Spec.To {
+	// 	// When empty, the Kubernetes core API group is inferred.
+	// 	group := "core/v1"
+	// 	if to.Group != "" {
+	// 		group = string(to.Group)
+	// 	}
+
+	// 	toSelectors = append(toSelectors, groupKindToFieldSelector(schema.GroupKind{
+	// 		Group: group,
+	// 		Kind:  string(to.Kind),
+	// 	}))
+	// }
+
+	routes := r.getReferencePolicyObjectsFrom(refPolicy)
+
+	// TODO: match only routes with BackendRefs selectable by a
+	// ReferencePolicyTo instead of appending all routes. This seems expensive,
+	// so not sure if it would actually improve performance or not.
+	matches = append(matches, routes...)
+
+	// for _, route := range routes {
+	// 	routeMatched := false
+
+	// 	// TODO: should this use reflection to handle xRoute types? seems expensive
+	// 	for _, rule := range route.Spec.Rules {
+	// 		for range rule.BackendRefs {
+	// 			// Check if backendRef.BackendObjectReference is selectable by
+	// 			// the requirements in any refPolicy.Spec.To
+	// 			for _, selector := range toSelectors {
+	// 				if selector.Matches(fields.Set{
+	//                  "kind": backendRef.BackendObjectReference.Kind
+	//                  "metadata.name": backendRef.BackendObjectReference.Name
+	// 				}) {
+	// 					routeMatched = true
+	// 					matches = append(matches, route)
+
+	// 					// Exit toSelectors loop early if route has already been matched
+	// 					if routeMatched {
+	// 						break
+	// 					}
+	// 				}
+	// 			}
+
+	// 			// Exit BackendRefs loop early if route has already been matched
+	// 			if routeMatched {
+	// 				break
+	// 			}
+	// 		}
+
+	// 		// Exit Rules loop early if route has already been matched
+	// 		if routeMatched {
+	// 			break
+	// 		}
+	// 	}
+	// }
+
+	return matches
+}
+
+func (r *HTTPRouteReconciler) getReferencePolicyObjectsFrom(refPolicy gateway.ReferencePolicy) []gateway.HTTPRoute {
+	matches := []gateway.HTTPRoute{}
+
+	for _, from := range refPolicy.Spec.From {
+		// TODO: search by from.Group and from.Kind instead of assuming HTTPRoute
+		routes, err := r.Client.GetHTTPRoutesInNamespace(context.TODO(), string(from.Namespace))
```

Comment:
you may want to pass the top-level context and store it on the reconciler struct so you can reference it here, just so we get clean cancelation of any outstanding requests during shutdown.